### PR TITLE
interface: UX renovations for new groups

### PR DESCRIPTION
**Pull request by matildepark**
Screen Shot 2020-07-15 at 4 16 34 PM

- Fixes a bug where invite search would return valid patps regardless of what `shipResults` was set to.
- Fixes a bug where Groups autojoin wasn't receiving props from the route and still had some cruft from its Publish scaffolding code.
- Drastically simplifies new Chat flows — 'New Chat' requires a group and links to Groups to create new ones; 'New DM' adds the ability to name and create group DMs and is graphically separated in the sidebar.
- Removes "Create group" toggle from all "New..." flows.
- Adds "Create group" link to Groups above the Invite Search input on New screens.
- Removes "Join [chat,notebook]" from sidebars.
- Removes shortcode references from Chat and Publish.
- Adds a link to join Urbit Community directly from the Launch welcome message.
- Restores shortcode links in Chat — they now link to Groups and automatically join the group (if the group is public).

### CODE REVIEW

File: pkg/interface/src/apps/chat/components/sidebar.js [hunk 1]
```diff
@@ -109,29 +86,21 @@ export class Sidebar extends Component {
           />
         );
       });
-      if (groupedChannels['/~/'] && groupedChannels['/~/'].length !== 0) {
+      if (groupedChannels['dm'] && groupedChannels['dm'].length !== 0) {
```
[the remark this answers is not in the document]
Comment (matildepark) [reply pkg/interface/src/apps/chat/components/sidebar.js:89]: Lol true. we can add a link to the landing page but it doesn’t show in mobile.

[the remark this answers is not in the document]
Comment (matildepark) [reply pkg/interface/src/apps/chat/components/sidebar.js:89]: We could show mobile only landing text for the sidebar if chats are empty.

[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/interface/src/apps/chat/components/sidebar.js:89]: I'd probably lean in favour of just removing this condition altogether tbh

[the remark this answers is not in the document]
Comment (matildepark) [reply pkg/interface/src/apps/chat/components/sidebar.js:89]: To just always show "Direct Messages"?

[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/interface/src/apps/chat/components/sidebar.js:89]: yeah

**Comment by matildepark** [comment]
Added quality of life changes for chat as per @urcades request.

Screen Shot 2020-07-16 at 2 14 48 PM

Screen Shot 2020-07-16 at 2 13 27 PM

**Comment by urcades** [comment]
beautiful!

—
~fabled-faster
https://urbit.org/

**Pull request merged**